File: restdocs-swagger/src/main/java/com/example/restdocsswagger/controller/SampleController.java
package com.example.restdocsswagger.controller;

import org.springframework.http.HttpStatus;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.ResponseStatus;
import org.springframework.web.bind.annotation.RestController;

@RestController
public class SampleController {

    @PostMapping("/users")
    @ResponseStatus(HttpStatus.OK)
    public SampleResponse returnNameAndAge(@RequestBody SampleRequest request) {
        return new SampleResponse(request.name(), request.age());
    }
}

File: k6-performance-test/src/main/kotlin/me/jinan159/sampleapi/SampleController.kt
package me.jinan159.sampleapi

import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/hello")
class SampleController {
    @GetMapping
    fun world(): Map<String, String> {
        return mapOf(
            "hello" to "world"
        )
    }
}